.changes/README.md

```diff
@@ -0,0 +1,46 @@
+# Changes
+
+This stores changes to automatically generate the changelog, to avoid merge conflicts. Files should be in a JSON format, with the following format:
+
+```json
+{
+    "description": "single-line description to add to the CHANGELOG.",
+    "issues": [894],
+    "type": "added",
+    "breaking": false
+}
+```
+
+Valid types are:
+- added (Added)
+- changed (Changed)
+- fixed (Fixed)
+- removed (Removed)
+- internal (Internal)
+
+`breaking` is optional and defaults to false. if `breaking` is present for any active changes, we will not allow a minor (or pre-1.0) patch version to be generated. `issues` is also optional, and is currently unused, and is an array of issues fixed by the PR, and defaults to an empty array.
+
+The file numbers should be `${pr}.json`
+
+If multiple changes are made in a single PR, you can also pass an array of entries:
```

Comment:
> I'd like it to be possible to have multiple prs on one line, and also an entry without pr number. e.g like the two below entries in 0.2.2
> 
> ```
> - #767, #788 - added the `cross-util` and `xtask` commands.
> - Change rust edition to 2021 and bump MSRV for the cross binary to 1.58.1
> ```
> 
> this could be implemented by naming the file `767-788.json` or similar, and for regular entries without a pr attached `_901.json` maybe?

That should definitely be doable, we just need a consistent naming convention. I guess we'd have to make a sort key and then make the PR optional, which could be an option or an issue number, and one must be present.

Reply:
This has been fixed with the following: entries with the format `issue${issue}.json` won't have an associated PR number, and the rest will. For example, a breaking change without a PR attached would produce something like `- BREAKING: this is one added entry.`.

It supports the following syntax for PRs:

- `${pr1}-${pr2}(...).json`, which will be output as `- #979,981 - this has 2 PRs associated.`
- `${issue1}-${issue2}.json`, which will be output as `- this is one added entry.`

Both PRs and issues can have 1 or more associated numbers with it, and both can have an object or an array objects for each entry. See the unittests for extensive documentation on this.